File: responsibility/interface-segregation/Payment.kt
interface PaymentProvider {
  fun paymentName()
  fun amount()
}

interface EWalletProvider {
  fun accountID()
  fun walletProviderID()
}

interface BankProvider {
  fun bankId()
  fun virtualAccount()
}

class Gopay : EWalletProvider, PaymentProvider {
  override fun paymentName() {
      // Implementation code
  }
  override fun amount() {
      // Implementation code
  }
  override fun accountID() {
      // Implementation code
  }
  override fun walletProviderID() {
      // Implementation code
  }
}

class Mandiri : BankProvider, PaymentProvider {
  override fun paymentName() {
      // Implementation code
  }
  override fun amount() {
      // Implementation code
  }
  override fun bankID() {
      // Implementation code
  }
  override fun virtualAccount() {
      // Implementation code
  }
}

class BNI : BankProvider, PaymentProvider {
  override fun paymentName() {
      // Implementation code
  }
  override fun amount() {
      // Implementation code
  }
  override fun bankID() {
      // Implementation code
  }
  override fun virtualAccount() {
      // Implementation code
  }
}

File: responsibility/interface-segregation/payment.dart
class EWalletProvider {
  void accountID() {}
  void walletProviderID() {}
}

class PaymentProvider {
  void paymentName() {}
  void amount() {}
}

class BankProvider {
  void bankID() {}
  void virtualAccount() {}
}

class Gopay implements EWalletProvider, PaymentProvider {
  @override
  void paymentName() {
    // implementation
  }

 @override
  void amount() {
    // implementation
  }

  @override
  void accountID() {
    // implementation
  }
  @override
  void walletProviderID() {
    // implementation
  }
}

class Mandiri implements BankProvider, PaymentProvider {
  @override
  void paymentName() {
    // implementation
  }

  @override
  void amount() {
    // implementation
  }

  @override
  void bankID() {
    // implementation
  }
  @override
  void virtualAccount() {
    // implementation
  }
}

class BNI implements BankProvider, PaymentProvider {
  @override
  void paymentName() {
    // implementation
  }

  @override
  void amount() {
    // implementation
  }

  @override
  void bankID() {
    // implementation
  }
  @override
  void virtualAccount() {
    // implementation
  }
}
